### PR TITLE
W-13851546 caching policy

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+NOTE: The `run` command starts a Flex Gateway instance inside a Docker container. By default, Docker containers use UTC as the local time. 
+
+. Send requests to Flex Gateway by using the following command as example:
++
+[source,ssh]
+----
+curl http://127.0.0.1:8081/catalog/1 -H "cache_check: cache_value"
+----
++
+If the request is sent during the configured non-business hours, the response contains the original header. If the request is sent during the configured business hours, the response is empty.
```
Comment (JaatadiaMulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: Since the backend is an echo server, If you send a second request with a different header value during the configured non-business hours, youll have the  header of the first request since it was cached. if not the header of the second request will be present

**Review by JaatadiaMulesoft — CHANGES_REQUESTED** [review]

**Review by JaatadiaMulesoft — APPROVED** [review]

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+Use the data caching policy template as an example of how to xref:policies-pdk-configure-features.adoc#share-data-between-workers-and-caching.adoc[configure policy caching].
+
+== Before You Begin
+
+. xref:policies-pdk-prerequisites.adoc[]
+
+== Policy Use Case
+
+An antique store that buys and sells items needs to track the stock of each item in its collection and who buys it. To track the inventory movement, the company uses an API where each unique path represents a different item..
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]:
```suggestion
An antique store that buys and sells items needs to track the stock of each item in its catalog and who buys it. To track the inventory movement, the company uses an API where each unique path represents a different item.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+
+. xref:policies-pdk-prerequisites.adoc[]
+
+== Policy Use Case
+
+An antique store that buys and sells items needs to track the stock of each item in its collection and who buys it. To track the inventory movement, the company uses an API where each unique path represents a different item.. 
+
+During business hours, the catalog changes constantly and caching the available items isn't very useful. Outside of business hours, the stock is frozen, and the catalog can be cached without any concern. Given this, the manager of the antique store's server decides to implement a caching policy that caches the stock during the non-business hours. 
+
+The policy first checks the time of each incoming request. Then, if the request is outside of business hours and if a response for the path has not been previously cached, the policy caches the response. Then the policy returns the cached response for all following requests to the path outside of business hours. When business hours resume, the cache is cleared and is ready to cache new responses during the next period of non-business hours.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: How about putting this is a more scannable format? Since it's essentially steps, you could put it something like the following:

The server manager implements the caching policy as follows:

. The policy checks the time of each incoming request. 
. If the request is outside of business hours and if a response for the path has not been previously cached, the policy caches the response. 
. The policy returns the cached response for all following requests to the path outside of business hours. 
. When business hours resume, Anypoint Platform clears the cache and is ready to cache new responses during the next period of non-business hours.

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+
+== Before You Begin
+
+. xref:policies-pdk-prerequisites.adoc[]
+
+== Policy Use Case
+
+An antique store that buys and sells items needs to track the stock of each item in its collection and who buys it. To track the inventory movement, the company uses an API where each unique path represents a different item.. 
+
+During business hours, the catalog changes constantly and caching the available items isn't very useful. Outside of business hours, the stock is frozen, and the catalog can be cached without any concern. Given this, the manager of the antique store's server decides to implement a caching policy that caches the stock during the non-business hours.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]:
```suggestion
During business hours, the catalog changes constantly and caching the available items isn't very useful. Outside of business hours, the stock is frozen, and the catalog can be cached without any concern. Given this, the antique store's server manager decides to implement a caching policy that caches the stock during the non-business hours. 
```

Comment (bjoson-mulesoft) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: I'm not sure about "isn't very useful" and "Given this" for global language and conciseness.
Could you simplify something like:
Because the store's catalog items in stock change constantly during business hours but are frozen outside of business hours, the server manager decides to implement a caching policy that caches the stock during non-business hours.

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+
+During business hours, the catalog changes constantly and caching the available items isn't very useful. Outside of business hours, the stock is frozen, and the catalog can be cached without any concern. Given this, the manager of the antique store's server decides to implement a caching policy that caches the stock during the non-business hours. 
+
+The policy first checks the time of each incoming request. Then, if the request is outside of business hours and if a response for the path has not been previously cached, the policy caches the response. Then the policy returns the cached response for all following requests to the path outside of business hours. When business hours resume, the cache is cleared and is ready to cache new responses during the next period of non-business hours.
+
+To reuse the policy, the amount of cached requests and the non-business hours are configurable. 
+
+An error in the caching flow should not make a request fail. So, by default, the caching policy does not block requests.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]:
```suggestion
An error in the caching flow should not make a request fail. By default, the caching policy does not block requests.  
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+NOTE: The `run` command starts a Flex Gateway instance inside a Docker container. By default, Docker containers use UTC as the local time. 
+
+. Send a request to Flex Gateway:
++
+[source,ssh]
+----
+curl http://127.0.0.1:8081/catalog/1 -H "cache_check: cache_value"
+----
++
+NOTE: The upstream service used in the debugging environment included with PDK responds with an echo of the request.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc:418]: Is PDK being used without "the"? Is PDK the official short name?

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+
+. Send another request to Flex Gateway changing the included header:
++
+[source,ssh]
+----
+curl http://127.0.0.1:8081/catalog/1 -H "cache_check: cache_value1"
+----
++
+If the requests were made outside of business hours, the header `cache_check: cache_value` is included in the response body instead of `cache_check: cache_value1`. If the requests were made inside of business hours, the header `cache_check: cache_value1` is included in the response body.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]:
```suggestion
If requests are made outside of business hours, the header `cache_check: cache_value` is included in the response body instead of `cache_check: cache_value1`. If requests are made inside of business hours, the header `cache_check: cache_value1` is included in the response body.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+
+. Adjust the configured non-business hours and the request path to view different responses.
```
Comment (bjoson-mulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]:
```suggestion
. Change the non-business hours and request path to view different responses.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc (hunk 7 above)
Comment (bjoson-mulesoft) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: Logic: Avoid past tense

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc (hunk 8 above)
Comment (bjoson-mulesoft) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: Logic: Use the simpler term (change) and simplify the wording. Wouldn't "the configured" be implied, especially in context?

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc (hunk 2 above)
Comment (bjoson-mulesoft) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc]: I suggest you use either collection and catalog to be consistent. 
Same with inventory and stock

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc (hunk 6 above)
Comment (glenn-rodgers-sf) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc:418]: I'd say PDK is the official short name, but I'm not exactly sure. Let me talk to the PM and then I will update these all universally later.

**Review by bjoson-mulesoft — APPROVED** [review]
